src/workerd/jsg/jsvalue.h

```diff
@@ -0,0 +1,467 @@
+  using JsBase<v8::Date, JsDate>::JsBase;
+};
+
+#define V(Name) \
+  class Js##Name final : public JsBase<v8::Name, Js##Name> { \
+  public: \
+    using JsBase<v8::Name, Js##Name>::JsBase; \
+  };
+
+  V(Symbol)
+  V(BigInt)
+  V(Number)
+  V(Int32)
+  V(Uint32)
+  V(Map)
+  V(Set)
+
+#undef V
+
+class JsObject final : public JsBase<v8::Object, JsObject> {
+public:
+  operator JsMap() const;
+
+  void set(Lock& js, const JsValue& name, const JsValue& value);
+  void set(Lock& js, kj::StringPtr name, const JsValue& value);
+  JsValue get(Lock& js, const JsValue& name) KJ_WARN_UNUSED_RESULT;
+  JsValue get(Lock& js, kj::StringPtr name) KJ_WARN_UNUSED_RESULT;
+
+  enum class HasOption {
+    NONE,
+    OWN,
+  };
+
+  bool has(Lock& js, const JsValue& name, HasOption option = HasOption::NONE) KJ_WARN_UNUSED_RESULT;
+  bool has(Lock& js, kj::StringPtr name, HasOption option = HasOption::NONE) KJ_WARN_UNUSED_RESULT;
+  void delete_(Lock& js, const JsValue& name);
+  void delete_(Lock& js, kj::StringPtr name);
+
+  void setPrivate(Lock& js, kj::StringPtr name, const JsValue& value);
+  JsValue getPrivate(Lock& js, kj::StringPtr name) KJ_WARN_UNUSED_RESULT;
+  bool hasPrivate(Lock& js, kj::StringPtr name) KJ_WARN_UNUSED_RESULT;
+
+  int hashCode() const;
+
+  kj::String getConstructorName() KJ_WARN_UNUSED_RESULT;
+
+  using JsBase<v8::Object, JsObject>::JsBase;
+
+  explicit JsObject(v8::Local<v8::Map> inner);
+};
+
+template <typename T>
+inline kj::Maybe<T> JsValue::tryCast() const {
+  if constexpr (kj::isSameType<T, JsValue>()) { return JsValue(inner); }
+#define V(Name) \
+  else if constexpr (kj::isSameType<T, Js##Name>()) { \
+    if (!inner->Is##Name()) return nullptr; \
+    return T(inner.template As<v8::Name>()); \
+  }
+  JS_TYPE_CLASSES(V)
+#undef V
+  else { return nullptr; }
+}
+
+template <typename T>
+inline kj::Maybe<T&> JsValue::tryGetExternal(Lock& js, const JsValue& value) {
+  if (!value.isExternal()) return nullptr;
+  return kj::Maybe<T&>(*static_cast<T*>(value.inner.As<v8::External>()->Value()));
+}
+
+template <typename T>
+inline kj::Array<T> JsString::toArray(Lock& js, WriteOptions options) const {
+  if constexpr (kj::isSameType<T, kj::byte>()) {
+    KJ_ASSERT(inner->ContainsOnlyOneByte());
+    auto buf = kj::heapArray<kj::byte>(inner->Length());
+    inner->WriteOneByte(js.v8Isolate, buf.begin(), 0, buf.size(), options);
+    return kj::mv(buf);
+  } else {
+    auto buf = kj::heapArray<uint16_t>(inner->Length());
+    inner->Write(js.v8Isolate, buf.begin(), 0, buf.size(), options);
+    return kj::mv(buf);
+  }
+}
+
+template <typename...Args, typename>
+inline JsArray Lock::arr(const Args&... args) {
+  v8::Local<v8::Value> values[] = { args... };
+  return JsArray(v8::Array::New(v8Isolate, &values[0], sizeof...(Args)));
+}
+
+template <typename...Args, typename>
+inline JsSet Lock::set(const Args&...args) {
+  auto set = v8::Set::New(v8Isolate);
+  check(set->Add(v8Context(), args.inner...));
```

Comment:
I don't think `Set::Add()` is variadic: https://v8.github.io/api/head/classv8_1_1Set.html#afe7b3f75000e7bca468199e87cea27b7

I think you want:
```cpp
  (check(set->Add(v8Context(), args.inner)), ...);
```